File: trunk/js/org/jsmol/test/Test_bool.js
Clazz.declarePackage ("org.jsmol.test");
Clazz.load (null, "org.jsmol.test.Test_bool", ["java.lang.Boolean", "java.util.Hashtable"], function () {
c$ = Clazz.declareType (org.jsmol.test, "Test_bool");
c$.main = Clazz.defineMethod (c$, "main", 
function (args) {
 new org.jsmol.test.Test_bool ();
}, "~A");
Clazz.makeConstructor (c$, 
function () {
var bTrue = Boolean.TRUE;
var bFalse = Boolean.FALSE;
System.out.println ("bTrue.booleanValue() is " + bTrue.booleanValue ());
System.out.println ("bFalse.booleanValue() is " + bFalse.booleanValue ());
var ht =  new java.util.Hashtable ();
ht.put ("True", bTrue);
ht.put ("False", bFalse);
System.out.println ("bTrue in ht is " + ht.get ("True"));
System.out.println ("bTrue.booleanValue() in ht is " + (ht.get ("True")).booleanValue ());
System.out.println ("bFalse in ht is " + ht.get ("False"));
System.out.println ("bFalse.booleanValue() in ht is " + (ht.get ("False")).booleanValue ());
});
});

File: src/org/jsmol/test/Test_bool.java
package org.jsmol.test;

import java.util.Hashtable;
import java.util.Map;

public class Test_bool {

  // Main application
  public static void main(String[] args) {
   new Test_bool();
 }

	public Test_bool() {
		Boolean bTrue = Boolean.TRUE;
		Boolean bFalse = Boolean.FALSE;
		System.out.println("bTrue.booleanValue() is " + bTrue.booleanValue());
		System.out.println("bFalse.booleanValue() is " + bFalse.booleanValue());
		Map<String, Object>ht = new Hashtable<String, Object>();
		ht.put("True", bTrue);
		ht.put("False", bFalse);
		System.out.println("bTrue in ht is " + ht.get("True"));
		System.out.println("bTrue.booleanValue() in ht is " + ((Boolean)ht.get("True")).booleanValue());
		System.out.println("bFalse in ht is " + ht.get("False"));
		System.out.println("bFalse.booleanValue() in ht is " + ((Boolean)ht.get("False")).booleanValue());
	}

}

